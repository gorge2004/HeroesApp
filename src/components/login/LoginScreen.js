import { useNavigate } from "react-router-dom";
import { types } from '../../types/types';
import { useContext } from 'react';
import { AuthContext } from '../../auth/authContext';

function LoginScreen() {
  const navigate = useNavigate();
  const { user, dispatch} = useContext(AuthContext);

  const handleLogin = () => {
    const action = {
      type: types.login,
      payload : {
        name: "Jorge",
      }
    };
     dispatch(action);
     const  lastPath = localStorage.getItem('lastPath') || '/marvel';
     navigate(lastPath, { replace: true });
   
  };
  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />

      <button className="btn btn-primary" onClick={handleLogin}>
        Login
      </button>
    </div>
  );
}

export default LoginScreen;
