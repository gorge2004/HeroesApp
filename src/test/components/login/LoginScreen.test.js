import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";
import LoginScreen from "../../../components/login/LoginScreen";
import { types } from "../../../types/types";
const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));
describe("testing_login_component", () => {
  test("should _render_login_component", () => {
    const dispatch = jest.fn();
    const contextValue = { logged: false, dispatch };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={["/login"]}>
          <LoginScreen />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("button").text().trim()).toBe("Login");
  });
  test("should _calls_login_action", () => {
    const dispatch = jest.fn();
    const contextValue = { logged: false, dispatch };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter initialEntries={["/login"]}>
          <LoginScreen />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("button").text().trim()).toBe("Login");

    wrapper.find("button").simulate("click");
    const action = {
        type: types.login,
        payload : {
          name: "Jorge",
        }
      };
    expect(dispatch).toHaveBeenCalledWith(action);
    expect(mockNavigate).toHaveBeenCalledWith('/marvel', {replace: true});
    localStorage.setItem('lastPath','/dc');
    wrapper.find("button").simulate("click");
    expect(mockNavigate).toHaveBeenCalledWith('/dc', {replace: true});


  });
});
