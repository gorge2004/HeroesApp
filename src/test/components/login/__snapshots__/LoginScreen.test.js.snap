// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing_login_component should _calls_login_action 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`testing_login_component should _render_login_component 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
